From af2df16f733f93fb94a7418876d64cc046d66d8c Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Wed, 21 Feb 2018 22:36:23 +0100
Subject: [PATCH 149/173] Snap: Convert "save best" dialog text to a quantity
 string

* Saving 1 vs saving several can be a different string in other languages
 (e.g. German: "1 von 10 Bildern wird gespeichert[...]"
  vs "2 von 10 Bildern werden gespeichert[...]")
* Also fixes a missing "be" in the source string ("will _be_ deleted")

Change-Id: I0f0a91fa3cd1abe17c9d3eef453c4ed706a6fb25
---
 res/values/strings.xml                          | 13 +++++++++++--
 src/com/android/camera/BestpictureActivity.java |  7 ++++---
 2 files changed, 15 insertions(+), 5 deletions(-)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index bcf20bf44..f57919372 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -709,11 +709,20 @@ CHAR LIMIT = NONE] -->
     <string name="pref_camera2_restore_default">Restore defaults</string>
     <string name="pref_camera2_restore_default_hint">Camera settings will be restored to default settings</string>
 
-    <string name="save_best_dialog_content">%1$d of 10 images will be saved and the rest will deleted.</string>
+    <plurals name="save_best_dialog_content">
+        <item quantity="one">%1$d of 10 images will be saved and the rest will be deleted</item>
+        <item quantity="other">%1$d of 10 images will be saved and the rest will be deleted</item>
+    </plurals>
+
     <string name="save_best_dialog_title">Save selected images?</string>
     <string name="save_best_dialog_positive_bt">OK</string>
     <string name="save_best_dialog_native_bt">Cancel</string>
-    <string name="save_best_image_toast">%1$d of 10 images saved.</string>
+
+    <plurals name="save_best_image_toast">
+        <item quantity="one">%1$d of 10 images saved.</item>
+        <item quantity="other">%1$d of 10 images saved.</item>
+    </plurals>
+
     <string name="dialog_do_not_show">Do not show again</string>
     <string name="delete_best_dialog_title">Delete all?</string>
     <string name="delete_best_dialog_content">Are you sure you want to delete all 10 pictures taken?</string>
diff --git a/src/com/android/camera/BestpictureActivity.java b/src/com/android/camera/BestpictureActivity.java
index 6eaa2a3d0..e4c1ca073 100644
--- a/src/com/android/camera/BestpictureActivity.java
+++ b/src/com/android/camera/BestpictureActivity.java
@@ -411,7 +411,8 @@ public class BestpictureActivity extends FragmentActivity {
 
             @Override
             public String getContentString() {
-                return getResources().getString(R.string.save_best_dialog_content, choosenCount);
+                return getResources().getQuantityString(R.plurals.save_best_dialog_content,
+                    choosenCount, choosenCount);
             }
 
             @Override
@@ -496,8 +497,8 @@ public class BestpictureActivity extends FragmentActivity {
                 }
             }
         }
-        String toastString = getResources().getString(R.string.save_best_image_toast,
-        toSaveCount);
+        String toastString = getResources().getQuantityString(R.plurals.save_best_image_toast,
+                toSaveCount, toSaveCount);
         Toast.makeText(BestpictureActivity.this, toastString, Toast.LENGTH_SHORT).show();
         backToViewfinder();
     }
-- 
2.17.1

